tiled/adapters/sql.py

```diff
@@ -325,9 +285,12 @@ def append_partition(
             c: c.lower() for c in table.column_names if c != c.lower()
         }:
             table = table.rename_columns(upr_lwr_case_mapping)
-        with self.conn.cursor() as cursor:
+
+        conn = self.storage.connect()
+        with conn.cursor() as cursor:
             cursor.adbc_ingest(self.table_name, table, mode="append")
-        self.conn.commit()
+        conn.commit()
+        conn.close()
```

Comment:
Can a context manager or a `finally` block be used to ensure that this runs even if `commit()` raises? (See also just below, same issue.)